File: src/components/Events/Event.ts
import { Timestamp } from "firebase/firestore";

export interface Links {
  name: string;
  url: string;
}

export interface Event {
  date: Timestamp;
  title: string;
  description: string;
  img: string;
  register: string;
  links: Links[];
  time: string;
  venue: string;
  isActive: boolean;
  id: string;
  note: string;
}

File: firebase_py/Event.py

import datetime
import string

class Event:
    # def __init__(self, title:string, desc:string, img:string, is_active:bool, register:string, date:datetime, time:string, venue:string ):
    def __init__(self):
        self.title=""
        self.desc=""
        self.img=""
        self.is_active=False
        self.links=[]
        self.register=""
        self.date=datetime.datetime(2022, 1, 1)  # year, month, day
        self.time = ""
        self.venue=""
      